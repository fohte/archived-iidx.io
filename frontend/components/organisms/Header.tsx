import { Menu } from 'antd'
import Link from 'next/link'
import styled from 'styled-components'

import Brand from '@app/components/atoms/Brand'
import UserNav from '@app/components/molecules/UserNav'
import withAuthState from '@app/lib/withAuthState'

const Header: React.SFC = () => (
  <Menu theme="dark" style={{ lineHeight: '64px' }}>
    <Link href="/" prefetch>
      <BrandWrapper>
        <Brand />
      </BrandWrapper>
    </Link>
    <UserNavWrapper>
      <EnhancedUserNav />
    </UserNavWrapper>
  </Menu>
)

const BrandWrapper = styled.div`
  float: left;
`

const UserNavWrapper = styled.div`
  float: right;
`

const EnhancedUserNav = withAuthState()(UserNav)

export default Header
