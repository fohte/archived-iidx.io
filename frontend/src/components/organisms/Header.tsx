import * as React from 'react'
import { Container, Menu, Segment } from 'semantic-ui-react'

import Brand from '@app/components/atoms/Brand'
import UserNav from '@app/components/molecules/UserNav'
import withAuthState from '@app/lib/withAuthState'
import { Link } from '@app/routes'

const Header: React.SFC = () => (
  <Segment vertical inverted>
    <Menu secondary inverted>
      <Container>
        <Link route="/" prefetch passHref>
          <Menu.Item as="a">
            <Brand />
          </Menu.Item>
        </Link>
        <Menu.Menu position="right">
          <EnhancedUserNav />
        </Menu.Menu>
      </Container>
    </Menu>
  </Segment>
)

const EnhancedUserNav = withAuthState()(UserNav)

export default Header
