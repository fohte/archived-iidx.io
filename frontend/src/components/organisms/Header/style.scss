@import 'styles/colors.scss';

.header-nav {
  background-color: $header-bg-color;

  padding-bottom: 12px;
  padding-top: 12px;

  a {
    outline: none;
    text-decoration: none;
  }
}

.flex-container {
  display: flex;
  justify-content: space-between;
  align-items: center;

  min-height: 2em;
  margin: 0 1.5rem;
}

.item {
  justify-content: center;
  align-items: center;
  padding: 0.5em;

  color: $header-fg-color;

  user-select: none;

  &:hover,
  &:active {
    color: darkest-color($header-fg-color);
  }
}

.clickable {
  &:hover {
    cursor: pointer;
  }
}
