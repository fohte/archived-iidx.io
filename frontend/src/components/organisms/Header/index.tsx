import classnames from 'classnames/bind'
import * as React from 'react'

import Brand from '@app/components/atoms/Brand'
import Button from '@app/components/atoms/Button'
import Container from '@app/components/atoms/Container'
import UserMenu from '@app/components/molecules/UserMenu'
import AuthContext from '@app/contexts/AuthContext'
import { Link } from '@app/routes'
import * as css from './style.scss'

const cx = classnames.bind(css)

const Header: React.SFC = () => (
  <nav className={cx('headerNav')}>
    <Container>
      <div className={cx('flexContainer')}>
        <Link route="/" prefetch passHref>
          <a>
            <div className={cx('item')}>
              <Brand />
            </div>
          </a>
        </Link>
        <AuthContext.Consumer>
          {({ loading, viewer }) => {
            if (loading) {
              return <div className={cx('item')}>loading</div>
            }

            if (viewer) {
              return (
                <UserMenu
                  className={cx('item', 'clickable')}
                  displayName={viewer.name}
                />
              )
            }

            return (
              <div className={cx('item', 'non-padding', 'login-signup-group')}>
                <Link route="/login" prefetch>
                  <a>
                    <Button inverted color="white" size="small">
                      Login
                    </Button>
                  </a>
                </Link>
                <Link route="/signup" prefetch>
                  <a>
                    <Button size="small" color="white">
                      Sign Up
                    </Button>
                  </a>
                </Link>
              </div>
            )
          }}
        </AuthContext.Consumer>
      </div>
    </Container>
  </nav>
)

export default Header
