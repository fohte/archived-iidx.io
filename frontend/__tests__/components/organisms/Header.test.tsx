import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Header from '@app/components/organisms/Header'

describe('Header', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Header />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
