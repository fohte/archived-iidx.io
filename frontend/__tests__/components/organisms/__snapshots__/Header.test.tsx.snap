// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="headerNav"
>
  <div
    className="container"
  >
    <div
      className="flexContainer"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="item"
        >
          <span>
            iidx.io
          </span>
        </div>
      </a>
      <div
        className="item"
      >
        loading
      </div>
    </div>
  </div>
</nav>
`;
